src/pkgcore/ebuild/repo_objs.py

```diff
@@ -170,6 +171,217 @@ def __init__(self, metadata_xml, manifest):
         self.manifest = manifest
 
 
+class ProjectMember(object, metaclass=klass.generic_equality):
+    """Data on a single project member.
+
+    :type email: C{unicode} object
+    :ivar email: email address.
+    :type name: C{unicode} object or C{None}
+    :ivar name: full name
+    :type role: C{unicode} object or C{None}
+    :ivar role: role within the project.
+    :type is_lead: C{bool}
+    :ivar is_lead: whether the member is a project lead.
+    """
+
+    __slots__ = ('email', 'name', 'role', 'is_lead')
+    __attr_comparison__ = ('email', 'name', 'role', 'is_lead')
+
+    def __init__(self, email, name=None, role=None, is_lead=None):
+        if email is None:
+            raise ValueError('email for project member must not be null')
```

Comment:
Raising ValueError for each object during instantiation is fine, but I'd prefer then that it be caught, logged as an error, and the related entry skipped when parsing projects.xml if possible. Otherwise, an invalid projects.xml file will cause premature exits when a user possibly doesn't even care about projects.xml, e.g. pkgcheck will error out and throw a backtrace when running the related check when instead it should probably throw an invalid projects.xml warning and continue running the rest of the checks.

Along those lines, it could help to add some basic tests for invalid projects.xml to codify what should occur on certain types of bad input.

Reply:
Made it non-fatal and added extra tests. It kinda sucks for subproject indirection API but should be doable.